File: Coursera-Machine-Learning/Python-Code/ex2/plotData.py
import matplotlib.pyplot as plt
import numpy as np

def plotData(X, y):
    # Plots the data points with + for the positive exams
    # and o the negative exams. X is assumed to be a Mx2 matrix
    m = len(y)
    
    fig = plt.figure()
    
    # plot the positive and negative examples on a
    # 2D plot, using option 'k+' for the positive
    # examples and 'ko' for the negative examples
    pos_point = np.zeros((1, 2))
    neg_point = np.zeros((1, 2))
    
    # Separating neg point and pos point into 2 separated vector
    for i in range(m):
        if y[i] == 0:
            neg_point = np.vstack((neg_point, np.array([[X[i, 0], X[i, 1]]])))
        else:
            pos_point = np.vstack((pos_point, np.array([[X[i, 0], X[i, 1]]])))
    
    # Remove first [0. 0.] from two those vector
    neg_point = neg_point[1:, :]
    pos_point = pos_point[1:, :]
    
    plt.plot(neg_point[:, 0], neg_point[:, 1], 'yo')
    plt.plot(pos_point[:, 0], pos_point[:, 1], 'k+')

File: Coursera-Machine-Learning/Matlab-Code/machine-learning-ex2/ex2/plotData.m
function plotData(X, y)
%PLOTDATA Plots the data points X and y into a new figure 
%   PLOTDATA(x,y) plots the data points with + for the positive examples
%   and o for the negative examples. X is assumed to be a Mx2 matrix.

m = length(y);

% Create New Figure
figure;
hold on;

% ====================== YOUR CODE HERE ======================
% Instructions: Plot the positive and negative examples on a
%               2D plot, using the option 'k+' for the positive
%               examples and 'ko' for the negative examples.
%

pos_point = [NaN, NaN];
neg_point = [NaN, NaN];

% Separating neg point and pos point into 2 separated vector
for i = 1:m
  if (y(i) == 0)
    neg_point = [neg_point; X(i, 1), X(i, 2)];
  else
    pos_point = [pos_point; X(i, 1), X(i, 2)];
  endif
endfor

% Remove [NaN NaN] from two those vector
neg_point(1, :) = [NaN, NaN];
pos_point(1, :) = [NaN, NaN];

plot(neg_point(:, 1), neg_point(:, 2), 'or', 'markersize', 10, 'markerfacecolor', 'r');
plot(pos_point(:, 1), pos_point(:, 2), '+k', 'markersize', 10, 'linewidth', 2);

% Put some labels and legend
xlabel('Exam 1 score')
ylabel('Exam 2 score')

% Specified in plot order
legend('Admitted', 'Not admitted')
hold off;



% =========================================================================

end
